### PR TITLE
Enhance usage data documentation

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this?**
/kind documentation

**What does this PR do / why we need it**:
1.  Enhance documentation for usage data. Add the data collected for every command. See https://github.com/openshift/odo/commit/d3ff3e8268575097f2b9a512d973d85f5123e489
2. Add developer specific documentation for telemetry collection. See https://github.com/openshift/odo/commit/c60247231353a3f68a2cd5ae1611d3dd7189d757
3. Move all master branch links to the main branch. See https://github.com/openshift/odo/commit/10f701f89a6177fe7ea9ca89eeb2e02ec41d2c34

**Which issue(s) this PR fixes**:

Fixes part of #4462

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [X] Documentation 

- [ ] Update changelog

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: USAGE_DATA.adoc [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
 * command's ID
 * command's duration time
-* command's error message and error type (in case of failure)
+* command's pseudonymized error message and error type (in case of failure)
 * whether the command was run from a terminal
-* system information such as OS type
+* OS type
 * `odo` version in use
 
+Note that these commands does not include `--help` commands. We do not collect data about help commands.
```
Comment (feloy) [inline USAGE_DATA.adoc:16]:
```suggestion
Note that these commands do not include `--help` commands. We do not collect data about help commands.
```

File: docs/dev/usage-data.adoc [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        "cmdProperties": {
+            "componentType": "nodejs"
+        }
+    }
+}
+```
+
+Before sending the data to the server, error message is sanitized of any PII(Personally Identifiable Information) data which includes username, file paths, URLs, and pod exec commands executed via odo commands.
+
+Note that these commands does not include --help commands. We do not collect data about help commands.
```
Comment (feloy) [inline docs/dev/usage-data.adoc:51]:
```suggestion
Note that these commands do not include --help commands. We do not collect data about help commands.
```

**Comment by valaparthvi** [comment]
/retest

```
++ ./ci-firewall request --mainbranch main --sendqueue amqp.ci.queue.win.unit.send --sendtopic amqp.ci.topic.win.unit.send --sendexchange amqp.ci.exchange.win.unit.send --setupscript scripts/setup_script_unit.sh --jenkinsproject odo-windows-unit-pr-build --runscript scripts/run_script_unit.sh --timeout 4h00m
{"component":"entrypoint","file":"prow/entrypoint/run.go:165","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 2h0m0s timeout","severity":"error","time":"2021-06-09T03:29:01Z"}
{"component":"entrypoint","file":"prow/entrypoint/run.go:255","func":"k8s.io/test-infra/prow/entrypoint.gracefullyTerminate","level":"error","msg":"Process did not exit before 15s grace period","severity":"error","time":"2021-06-09T03:29:16Z"}
{"component":"entrypoint","error":"process timed out","file":"prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2021-06-09T03:29:16Z"}
error: failed to execute wrapped command: exit status 127
```

**Comment by valaparthvi** [comment]
/test psi-unit-test-windows

Reason - timeout

**Comment by valaparthvi** [comment]
/override ci/prow/psi-unit-test-windows

Reason - doc PR

**Pull request merged**